Filter activities when printing programme

carlobeltrame opened the pull request:
Fixes #3600 

- [x] Needs some UI guidance by @manuelmeister 
- [x] also allow filtering storycontext and safetyconsiderations
- [x] add a filter by day, to allow for inserting blank notes pages or other things in between program days
 
![image](https://github.com/user-attachments/assets/c841a07e-c546-426d-bec1-addfd6a3dc54)


Ideas for the future (as long as this is not reviewed, I might continue working on some of these):
- e2e test (hard because we can't test the contents of the pdfs yet)
- display a prose summary of the active filter (hard because of the way the filter UI components are built right now)
- allow to exclude a specific schedule entry (e.g. for perfectionists who want to hide the repeated schedule entries of the same activity, or for combined camps of Wolfsstufe and Pfadistufe, to print only the programme relevant for one Stufe)
- filter content nodes inside the printed activities (e.g. exclude notes) (quite clearly a different feature)

carlobeltrame committed:
Fix reactivity of schedule entry filters

carlobeltrame committed:
Move responsibility of loading endpoints

carlobeltrame committed:
Very basic version of filter UI

carlobeltrame committed:
Implement print filtering

carlobeltrame committed:
Extract component and fix none filter

carlobeltrame committed:
Only change filter value when closing dialog

carlobeltrame committed:
Indicate when filter is active

carlobeltrame committed:
Improve visuals and mobile UX

carlobeltrame committed:
Lint

carlobeltrame committed:
Always hide period filter in print configurator

The period selection can already be done separately when printing

carlobeltrame committed:
Fix and extend unit tests

carlobeltrame committed:
Wait for camp load in e2e test

manuelmeister commented:
Nice! My first reactions:
- As a user I want to see some feedback/impact of the filters. Maybe a simple result count could be enough for the mvp (already in the dialog). 
  <img width="908" alt="Bildschirmfoto 2025-05-18 um 18 01 53" src="https://github.com/user-attachments/assets/f1397f84-1f95-4516-afec-d905f6fd57fe" />


- The button on the "page" should probably be aligned at the bottom and the same button style as the actual filters on the dashboard/picasso. Also with correct active state.  
  <img width="423" alt="Bildschirmfoto 2025-05-18 um 17 57 45" src="https://github.com/user-attachments/assets/639957e4-13e4-4ca1-ac9a-91d687e98d4f" />

- Also the button could be like: `Filter YY activities` (and if filtered then `Filter: XX of YY activites` or as you described: `Filter: Kategorie & Status (XX)` or `Filter: LA und LS (XX)` if only category filter is active) 
- Additional filters/modifiers would be cool eventually and could perfectly fit in the dialog (just bikeshedding: hide empty activites, prevent printing activity fully for every schedule entry, …)

carlobeltrame committed:
Style dialog activator

carlobeltrame committed:
Add result counter to dialog

carlobeltrame committed:
Apply filter to ToC as well

manuelmeister committed:
Fix page preview layout in safari

manuelmeister commented on frontend/src/components/dialog/DialogForm.vue:
```diff
@@ -39,7 +39,7 @@
           </div>
 
-          <v-card-text>
+          <v-card-text v-if="$slots.error || error">
```
Nice had this fix in one of my closed PRs. Thanks!

manuelmeister reviewed (changes requested):
If I click on filter on the PagesConfig, each schedule entry gets loaded individually.

carlobeltrame committed:
Preload activities for filtering

carlobeltrame commented:
> If I click on filter on the PagesConfig, each schedule entry gets loaded individually.

Fixed in [a64bb23](https://github.com/ecamp/ecamp3/pull/7439/commits/a64bb237ef6a23a43865c92d56aef5cc48ced82e)

carlobeltrame committed:
Implement print filtering for activityList, story and safetyConsiderations

carlobeltrame committed:
Add filter by date

This could be useful in the future to print the programme day by day, in
order to insert notes pages or other pages in between. Right now, this
even enables a makeshift manual day overview with story and safety
considerations before each day of the programme.

carlobeltrame committed:
Extend tests

carlobeltrame committed:
Fix eternal skeleton loader

carlobeltrame committed:
Fix 'only my activities' button

carlobeltrame committed:
Fix visual bugs

carlobeltrame committed:
Revert changes to loadingEndpoints mechanism for now

carlobeltrame committed:
Fix tests after merge

carlobeltrame committed:
Implement picasso filtering
